airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/RouterService.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling;
+
+import io.airbyte.commons.temporal.scheduling.GeographyMapper;
+import io.airbyte.config.Geography;
+import io.airbyte.config.persistence.ConfigRepository;
+import jakarta.inject.Singleton;
+import java.io.IOException;
+import java.util.UUID;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Decides which Task Queue should be used for a given connection's sync operations, based on the
+ * configured {@link Geography}
+ */
+@Singleton
+@Slf4j
+@AllArgsConstructor
```

Comment:
Nit:  Our recommendation is to not use Lombok and instead create an explicit constructor with the beans to be injected.  This allows us to use annotations on the fields should that become necessary.

Reply:
@jdpgrailsdev for my knowledge, what does 'allows us to use annotations on the fields should that become necessary' mean?

(I understand and agree on not mixing constructor annotations for clarity)

Reply:
@davinchia With the Lombok generated constructor, there is no code for us to modify if we end up needing to add a `@Qualifier` annotation to specify which version of a singleton to inject, nor is there the ability for us to use `@Value` or `@Property` to inject configuration values via the constructor.  Put another way, the Lombok generated constructor only works for constructor-based injection if we do not need to annotate any of the parameters in order for Micronaut to make the correct injection.

Reply:
Got it. Thanks for explaining.

I think we should also start phasing out Lombok in our system for clarity. Does Micronaut have an `@Slf4j` annotation equivalent? I believe these two are the main Lombok annotations we use.

Reply:
@davinchia No, not that I am aware of.  It provides the configuration of the logging system, but the use of a logger is up to the developer.  I'm not sure that the `@Slf4j` annotation really saves us a whole lot, as declaring a logger is a 1 line statement, much like the annotation.